services/server/server_linux.go

```diff
@@ -67,3 +69,8 @@ func apply(ctx context.Context, config *srvconfig.Config) error {
 func newTTRPCServer() (*ttrpc.Server, error) {
 	return ttrpc.NewServer(ttrpc.WithServerHandshaker(ttrpc.UnixSocketRequireSameUser()))
 }
+
+func applyBoltOpenOptions(options *bolt.Options) {
+	options.MmapFlags = syscall.MAP_POPULATE
+	options.Mlock = true
```

Comment:
Checked the CI error. The rootless looks like running in container. My local docker container env shows me the memlock limit is 64kb. 

I am not sure what it can brings to us. I think it should be configure

Reply:
Legit. I will split this part out of this PR.